### DISCUSSION TITLE
Feat: show/hide <Thought>'s for new CoT-finetuned models

### DISCUSSION THREAD

**Discussion by philogicae** [open-webui/open-webui#7445] [Ideas]
Open source CoT-finetuned models (such as [marco-o1](https://huggingface.co/AIDC-AI/Marco-o1)) are beginning to show up. They use a specific format: first think, then answer. We are missing a simple option to show/hide the thinking process (\<Thought>...\</Thought>) in the UI (Ideally just to configure how to display tags in the output).
An example will be more explicit:

![IMG_20241127_005425_861](https://github.com/user-attachments/assets/68deb41c-6641-496a-bf97-ca382056b847)

**Comment by kim-gtek** [comment]
https://openwebui.com/f/matthewh/reasoning_manifold

Use this

**Comment by silentoplayz** [comment]
Related PR - https://github.com/open-webui/open-webui/pull/7438

**Comment by DhavalThkkar** [comment]
Can we get one that supports azure openai based models?

**Comment by arunbugkiller** [comment]
@tjbck Hi, is there any way we can get to show thinking process on the front-end especially for models hosted via LiteLLM. Currently it goes blank on the front-end (Attached a screenshot of one such instance), although it does fetch results after a while but there is no visibility on the underlying process. I am using o3-mini model hosted via LiteLLM through azure AI. 

![Screenshot 2025-06-03 114843](https://github.com/user-attachments/assets/cb18bc61-7647-4ab3-a73b-7ffc3622c138)

**Comment by Cyp9715** [comment]
This feature is essential. CoT is generally not something most people need to see. It is enough to simply indicate that CoT is in progress. This appears to be configured correctly in models such as gpt-oss, but it does not seem to work properly with models like Qwen and Gemma.
